### PR TITLE
fix(storage): add bucket policy to force SSL connections

### PR DESCRIPTION
**Pull request by bvtujo**
This PR adds a bucket policy to the CF generated by `storage init`. It now requires SSL connections in order to perform any action on a bucket. 

Related #769 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.13
 
 require (
 	github.com/AlecAivazis/survey/v2 v2.0.7
-	github.com/Netflix/go-expect v0.0.0-20190729225929-0e00d9168667 // indirect
+	github.com/aws/amazon-ecs-cli-v2 v0.1.0 // indirect
```
Comment (iamhopaul123) [inline go.mod:7]: hmm why do we need this?

Comment (bvtujo) [reply go.mod:7]:
```bash
➜  oaas git:(fargate-efs) ✗ ./ecs storage init -t S3 -n new-ssl-bucket
Only found one service in workspace, defaulting to: fe
✔ Wrote CloudFormation template for S3 Bucket new-ssl-bucket at copilot/fe/addons/new-ssl-bucket.yml
The Cloudformation template is a nested stack which fully describes your resource,
the IAM policy necessary for an ECS task to access that resource, and outputs
which are injected as environment variables into the Copilot service this addon
is associated with.

Recommended follow-up actions:
- Update your service code to leverage the injected environment variable `NEWSSLBUCKET_NAME`
- Run `copilot svc deploy --name fe` to deploy your storage resources to your environments.
➜  oaas git:(fargate-efs) ✗ now
12:27:48
➜  oaas git:(fargate-efs) ✗ ./ecs svc deploy
Only found one service in workspace, defaulting to: fe
Only found one environment, defaulting to: test
Sending build context to Docker daemon  15.15MB
Step 1/6 : FROM golang:1.13
...
Successfully built 3928642169f3
Successfully tagged xxxxxxx.dkr.ecr.us-east-2.amazonaws.com/otter/fe:13d7972
Login Succeeded
The push refers to repository [xxxxxxx.dkr.ecr.us-east-2.amazonaws.com/otter/fe]
a4d36dbf8585: Pushed
f363be2e9e0f: Pushed

13d7972: digest: sha256:ab87990440fb3008edb411c99bee5dd94a6429eb838946539505e098b7350f15 size: 2425
➜  oaas git:(fargate-efs) ✗ date '+%X'
12:33:24
```

Looks like the whole update (including docker build & push) took 6 minutes. 

I also set up this service to write an object to S3 at container startup just to ensure that the bucket policy is not blocking all connections:
image


And finally, the rendered bucket policy as it appears on the bucket:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "ForceHTTPS",
            "Effect": "Deny",
            "Principal": "*",
            "Action": "s3:*",
            "Resource": [
                "arn:aws:s3:::otter-test-fe-s3-ssl/*",
                "arn:aws:s3:::otter-test-fe-s3-ssl"
            ],
            "Condition": {
                "Bool": {
                    "aws:SecureTransport": "false"
                }
            }
        }
    ]
}
```

**Comment by bvtujo** [comment]
```bash
➜  oaas git:(fargate-efs) ✗ ./ecs storage init -t S3 -n new-ssl-bucket
Only found one service in workspace, defaulting to: fe
✔ Wrote CloudFormation template for S3 Bucket new-ssl-bucket at copilot/fe/addons/new-ssl-bucket.yml
The Cloudformation template is a nested stack which fully describes your resource,
the IAM policy necessary for an ECS task to access that resource, and outputs
which are injected as environment variables into the Copilot service this addon
is associated with.

Recommended follow-up actions:
- Update your service code to leverage the injected environment variable `NEWSSLBUCKET_NAME`
- Run `copilot svc deploy --name fe` to deploy your storage resources to your environments.
➜  oaas git:(fargate-efs) ✗ now
12:27:48
➜  oaas git:(fargate-efs) ✗ ./ecs svc deploy
Only found one service in workspace, defaulting to: fe
Only found one environment, defaulting to: test
Sending build context to Docker daemon  15.15MB
Step 1/6 : FROM golang:1.13
...
Successfully built 3928642169f3
Successfully tagged xxxxxxx.dkr.ecr.us-east-2.amazonaws.com/otter/fe:13d7972
Login Succeeded
The push refers to repository [xxxxxxx.dkr.ecr.us-east-2.amazonaws.com/otter/fe]
a4d36dbf8585: Pushed
f363be2e9e0f: Pushed

13d7972: digest: sha256:ab87990440fb3008edb411c99bee5dd94a6429eb838946539505e098b7350f15 size: 2425
➜  oaas git:(fargate-efs) ✗ date '+%X'
12:33:24
```

Looks like the whole update (including docker build & push) took 6 minutes. 

I also set up this service to write an object to S3 at container startup just to ensure that the bucket policy is not blocking all connections:
image


And finally, the rendered bucket policy as it appears on the bucket:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "ForceHTTPS",
            "Effect": "Deny",
            "Principal": "*",
            "Action": "s3:*",
            "Resource": [
                "arn:aws:s3:::bucket/*",
                "arn:aws:s3:::bucket"
            ],
            "Condition": {
                "Bool": {
                    "aws:SecureTransport": "false"
                }
            }
        }
    ]
}
```

File: go.mod (hunk 1 above)
Comment (bvtujo) [reply go.mod]: Good point, I don't know why that keeps getting added to my go.mod, but more to the point It doesn't need to be in this PR.

**Pull request merged**